### PR TITLE
[Feature]Encode integers/binary per column for exchange

### PR DESCRIPTION
**Pull request by fzhedu**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

encode integers/binary per column for exchange, controlled by `transmission_encode_level`,
if transmission_encode_level & 2, intergers are encode by streamvbyte, in order or not;
if transmission_encode_level & 4, binary columns are compressed by lz4.
if transmission_encode_level & 1, enable adaptive encoding. 

e.g.,
if  transmission_encode_level = 7, SR will adaptively encode numbers and string columns according to the proper encoding ratio(< 0.9);
if  transmission_encode_level = 6, SR will force encoding numbers and string columns.

in short, for transmission_encode_level, 2 for encoding integers or types supported by integers, 4 for encoding string, json and object columns are left to be supported later.

### NOTE: 
to be compatible with older version, during downgrading/upgrading across this PR, `transmission_encode_level` must be 0 and waiting already running queries done, then replace binary.

## Problem Summary(Required) ：

encode integers for exchange to reduce data size but costing a little CPU.

### effects:

```
mysql> set transmission_encode_level =7;
mysql> select max(l_linenumber), max(l_orderkey),max(l_partkey),max(l_tax),max(l_discount),max(l_extendedprice),max(l_quantity),max(l_shipdate),min(l_suppkey),max(l_commitdate) from orders join lineitem on o_orderkey=l_partkey  where    l_orderkey< 500000000;
+-------------------+-----------------+----------------+------------+-----------------+----------------------+-----------------+-----------------+----------------+-------------------+
| max(l_linenumber) | max(l_orderkey) | max(l_partkey) | max(l_tax) | max(l_discount) | max(l_extendedprice) | max(l_quantity) | max(l_shipdate) | min(l_suppkey) | max(l_commitdate) |
+-------------------+-----------------+----------------+------------+-----------------+----------------------+-----------------+-----------------+----------------+-------------------+
|                 7 |       499999974 |       20000000 |       0.08 |            0.10 |            104798.50 |           50.00 | 1998-12-01      |              1 | 1998-10-31        |
+-------------------+-----------------+----------------+------------+-----------------+----------------------+-----------------+-----------------+----------------+-------------------+
1 row in set (4.32 sec)

mysql> set transmission_encode_level =0;
Query OK, 0 rows affected (0.00 sec)

mysql> select max(l_linenumber), max(l_orderkey),max(l_partkey),max(l_tax),max(l_discount),max(l_extendedprice),max(l_quantity),max(l_shipdate),min(l_suppkey),max(l_commitdate) from orders join lineitem on o_orderkey=l_partkey  where    l_orderkey< 500000000;
+-------------------+-----------------+----------------+------------+-----------------+----------------------+-----------------+-----------------+----------------+-------------------+
| max(l_linenumber) | max(l_orderkey) | max(l_partkey) | max(l_tax) | max(l_discount) | max(l_extendedprice) | max(l_quantity) | max(l_shipdate) | min(l_suppkey) | max(l_commitdate) |
+-------------------+-----------------+----------------+------------+-----------------+----------------------+-----------------+-----------------+----------------+-------------------+
|                 7 |       499999974 |       20000000 |       0.08 |            0.10 |            104798.50 |           50.00 | 1998-12-01      |              1 | 1998-10-31        |
+-------------------+-----------------+----------------+------------+-----------------+----------------------+-----------------+-----------------+----------------+-------------------+
1 row in set (5.48 sec)
```
## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
Can't set status; build succeeded.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

File: be/CMakeLists.txt [hunk 1]
```diff
@@ -241,6 +241,9 @@ set_target_properties(brotlienc PROPERTIES IMPORTED_LOCATION ${THIRDPARTY_DIR}/l
 add_library(zstd STATIC IMPORTED)
 set_target_properties(zstd PROPERTIES IMPORTED_LOCATION ${THIRDPARTY_DIR}/lib64/libzstd.a)
 
+add_library(streamvbyte STATIC IMPORTED)
+set_target_properties(streamvbyte PROPERTIES IMPORTED_LOCATION ${THIRDPARTY_DIR}/lib64/libstreamvbyte.a)
```
Comment (silverbullet233) [inline be/CMakeLists.txt:245]: remember to update `thirdparty/build-thirdparty.sh` and `thridparty/download-thirdparty.sh` if you import a new library

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 2]
```diff
@@ -585,6 +587,9 @@ public class SessionVariable implements Serializable, Writable, Cloneable {
     @VariableMgr.VarAttr(name = TRANSMISSION_COMPRESSION_TYPE)
     private String transmissionCompressionType = "LZ4";
 
+    @VariableMgr.VarAttr(name = TRANSMISSION_ENCODE_LEVEL)
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:590]: how many encode levels could we have? and what's the meaning for each level?

Comment (fzhedu) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:602]: encode_level & 1  for encode fixed_column
encode_level & 2 for encode binary_column

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

File: be/CMakeLists.txt (hunk 1 above)
Comment (fzhedu) [reply be/CMakeLists.txt]: done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|2|50.00%|[1316]|

File: thirdparty/build-thirdparty.sh [hunk 3]
```diff
@@ -357,7 +357,7 @@ build_snappy() {
     ${BUILD_SYSTEM} install
     if [ -f $TP_INSTALL_DIR/lib64/libsnappy.a ]; then
         mkdir -p $TP_INSTALL_DIR/lib
-        cp $TP_INSTALL_DIR/lib64/libsnappy.a $TP_INSTALL_DIR/lib/libsnappy.a
+        cp $TP_INSTALL_DIR/lib64/libsnappy.a $/lib/libsnappy.a
     fi
```
Comment (fzhedu) [inline thirdparty/build-thirdparty.sh:361]: delete by mistake

Comment (fzhedu) [reply thirdparty/build-thirdparty.sh]: done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|2|50.00%|[1316]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|2|50.00%|[1316]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|2|50.00%|[1316]|

**Comment by wangruin** [comment]
run starrocks_be_unittest image

**Comment by fzhedu** [comment]
run starrocks_be_unittest image

File: gensrc/thrift/InternalService.thrift [hunk 4]
```diff
@@ -193,6 +193,8 @@ struct TQueryOptions {
   65: optional bool enable_replicated_storage;
 
   66: optional bool use_scan_block_cache;
+
+  67: optional i32 transmission_encode_level;
```
Comment (mofeiatwork) [inline gensrc/thrift/InternalService.thrift:197]: conflicted

**Comment by fzhedu** [comment]
run starrocks_be_unittest image

File: gensrc/thrift/InternalService.thrift (hunk 4 above)
Comment (fzhedu) [reply gensrc/thrift/InternalService.thrift]: solved

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Comment by fzhedu** [comment]
run starrocks_be_unittest image

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Comment by fzhedu** [comment]
run starrocks_be_unittest image

File: be/src/serde/protobuf_serde.cpp [hunk 5]
```diff
@@ -11,16 +11,84 @@
...
+// if encode ratio < EncodeRatioLimit, encode it, otherwise not.
+void EncodeContext::_adjust(const int col_id) {
+    auto old_level = _column_encode_level[col_id];
+    if (_encoded_bytes[col_id] < _raw_bytes[col_id] * EncodeRatioLimit) {
+        _column_encode_level[col_id] = _session_encode_level;
+    } else {
+        _column_encode_level[col_id] = 0;
+    }
+    if (old_level != _column_encode_level[col_id] || _session_encode_level < -1) {
+        LOG(WARNING) << "Old encode level " << old_level << " is changed to " << _column_encode_level[col_id]
```
Comment (silverbullet233) [inline be/src/serde/protobuf_serde.cpp:49]: Is it more appropriate to use debug log here?

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java (hunk 2 above)
Comment (kangkaisen) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:607]: Add this info to comment and document.

File: be/src/serde/protobuf_serde.h [hunk 6]
```diff
@@ -14,29 +14,71 @@ class RowDescriptor;
 }
 
 namespace starrocks::serde {
+constexpr double EncodeRatioLimit = 0.9;
+constexpr uint32_t EncodeSamplingNum = 5;
+
+// EncodeContext adaptively adjusts encode_level according to the compression ratio. In detail,
+// for every _frequency chunks, if the compression ratio for the first EncodeSamplingNum chunks is less than
```
Comment (kangkaisen) [inline be/src/serde/protobuf_serde.h:21]: do we need to change the encode level multi times, not only once.

File: be/src/serde/protobuf_serde.h [hunk 7]
```diff
@@ -14,29 +14,71 @@ class RowDescriptor;
...
+    }
+
+    void set_encode_levels_in_pb(ChunkPB* const res);
+
+private:
+    // if encode ratio < EncodeRatioLimit, encode it, otherwise not.
+    void _adjust(const int col_id);
+    const int _session_encode_level;
+    uint64_t _times = 0;
+    uint64_t _frequency = 64;
```
Comment (kangkaisen) [inline be/src/serde/protobuf_serde.h:50]: what's the meaning of `_frequency` in here?

File: be/src/serde/column_array_serde.cpp [hunk 8]
```diff
@@ -53,41 +58,152 @@ const uint8_t* read_raw(const uint8_t* buff, void* target, size_t size) {
...
+    uint64_t encode_size = 0;
+    buff = read_little_endian_64(buff, &encode_size);
+    uint64_t encode_size1 = 0;
+    if (sorted_32ints) {
+        encode_size1 = streamvbyte_delta_decode(buff, (uint32_t*)target, (3 + size) * 1.0 / 4.0, 0);
+    } else {
+        encode_size1 = streamvbyte_decode(buff, (uint32_t*)target, (3 + size) * 1.0 / 4.0);
+    }
+    if (encode_size != encode_size1) {
+        throw std::runtime_error(fmt::format(
```
Comment (kangkaisen) [inline be/src/serde/column_array_serde.cpp:93]: why not return error status, but throw runtime error? will we meet these runtime_errors in user prod envs?

File: be/src/serde/protobuf_serde.h (hunk 6 above)
Comment (fzhedu) [reply be/src/serde/protobuf_serde.h:21]: yes, the encoding ratio may be changed for different chunks. we will check the encoding ratio every _frequency chunks.

File: be/src/serde/protobuf_serde.h (hunk 7 above)
Comment (fzhedu) [reply be/src/serde/protobuf_serde.h:50]: for every _frequency chunks, we will check the encoding ratio to decide encoding or not.

File: be/src/serde/protobuf_serde.cpp (hunk 5 above)
Comment (fzhedu) [reply be/src/serde/protobuf_serde.cpp:49]: I do not find `LOG(DEBUG)` in SR, can we use it?

File: be/src/serde/column_array_serde.cpp [hunk 9]
```diff
@@ -53,41 +58,152 @@ const uint8_t* read_raw(const uint8_t* buff, void* target, size_t size) {
...
+    VLOG_ROW << fmt::format("raw size = {}, encoded size = {}, integers compression ratio = {}\n", size, encode_size,
+                            encode_size * 1.0 / size);
+    return buff + encode_size;
+}
+
+template <bool sorted_32ints>
+const uint8_t* decode_integers(const uint8_t* buff, void* target, size_t size) {
+    uint64_t encode_size = 0;
+    buff = read_little_endian_64(buff, &encode_size);
+    uint64_t encode_size1 = 0;
```
[the remark this answers is not in the document]
Comment (fzhedu) [reply be/src/serde/column_array_serde.cpp:86]: changed to decode_size

File: be/src/serde/column_array_serde.cpp (hunk 8 above)
Comment (fzhedu) [reply be/src/serde/column_array_serde.cpp:93]: that will change return type of some functions, related a lot of codes.

if this error happens, it is fatal error, we should close the encoding function.

File: be/src/runtime/data_stream_sender.cpp [hunk 10]
```diff
@@ -450,6 +450,9 @@ Status DataStreamSender::prepare(RuntimeState* state) {
     _state = state;
     _be_number = state->be_number();
 
+    if (state->query_options().__isset.transmission_encode_level) {
```
[the remark this answers is not in the document]
Comment (fzhedu) [reply be/src/runtime/data_stream_sender.cpp:453]: removed

**Comment by fzhedu** [comment]
run starrocks_be_unittest image

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Comment by fzhedu** [comment]
run starrocks_be_unittest image

File: be/src/serde/column_array_serde.cpp [hunk 11]
```diff
@@ -53,41 +58,145 @@ const uint8_t* read_raw(const uint8_t* buff, void* target, size_t size) {
     return buff + size;
 }
 
-template <typename T>
+template <bool sorted_32ints>
+uint8_t* encode_integers(const void* data, size_t size, uint8_t* buff, int encode_level) {
```
Comment (LiShuMing) [inline be/src/serde/column_array_serde.cpp:62]: maybe add a benchmark in bench to compare with old codes?

**Review by murphyatwork — APPROVED** [review]

File: be/src/serde/column_array_serde.cpp (hunk 11 above)
Comment (fzhedu) [reply be/src/serde/column_array_serde.cpp:62]: there are some comprehensive benchmarks to compre various encoding/compression algorithms, https://github.com/powturbo/TurboPFor-Integer-Compression/issues/47,  so we can refer to it.

File: be/src/serde/protobuf_serde.cpp (hunk 5 above)
Comment (silverbullet233) [reply be/src/serde/protobuf_serde.cpp:49]: > I do not find `LOG(DEBUG)` in SR, can we use it?

maybe you need to use VLOG

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Review by silverbullet233 — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by fzhedu** [comment]
run starrocks_admit_test

**Comment by fzhedu** [comment]
run starrocks_admit_test

**Comment by fzhedu** [comment]
run starrocks_admit_test image

**Review by Seaven — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Pull request merged**